internal/backend/runtime/omni/state.go

```diff
@@ -90,7 +90,8 @@ func newNamespacedState(params *config.Params, primaryStorageCoreState state.Cor
 		return nil, nil, fmt.Errorf("failed to create etcd backup store: %w", err)
 	}
 
-	infraProviderState := infraprovider.NewState(primaryStorageCoreState, logger.With(logging.Component("infraprovider_state")))
+	infraProviderEphemeralState := inmem.NewStateWithOptions(inmem.WithHistoryGap(20))(resources.InfraProviderNamespace)
```

Comment:
I might understand the reasoning to use the same namespace for the ephemeral resources here, but it will be really hard to understand what's going on.
We route the resource to the ephemeral storage by it's type in the infra provider state.
But my proposal was to to use `resources.EphemeralNamespace` for the `InfraProviderHealthStatus` and just allow to write to that namespace.

Reply:
The current schema with the new inmem state can be still used, but I'd propose to use the ephemeral namespace.

I think namespaced state can be used to achieve that. You put that one on top, then ephemeral and etcd states underneath.

Reply:
Do you mean the infra provider state should be the "top-level" state now, and it should only catch/modify the infra provider related resources, and pass everything else through without touching them?

It doesn't feel right to me, because imo, the infra provider state should only get the resources it is interested in, not more.

If the issue with current approach is "looking at the resource, it is not obvious that it is ephemeral" (because it's namespace does not reflect that), then actually an easier solution would be to create a new ns called `infra-provider-ephemeral`. I also considered that as an option.

Reply:
One solution I also considered:
- introduce a new state wrapper called `RootState` let's say
- that is responsible of routing resources to the correct underlying state - not the "namespaced state" anymore
- it observes `namespace+type` to route the resources to the right underlying state
- the "infra provider state" is aware of both persistent and ephemeral states (passed via constructor)
- it does its intercepting logic and sends it to one of the underlying states

sth like:
```
root state
├── etcd state
├── ephemeral state
└── infra provider state
    ├── etcd state
    └── ephemeral state
```

Reply:
> Do you mean the infra provider state should be the "top-level" state now, and it should only catch/modify the infra provider related resources, and pass everything else through without touching them?

Not top level. It can be like it is right now.

I think I get the problem with using `ephemeral` namespace name.
Yeah, we can `infra-provider-ephemeral` namespace name and then make the infra provider handle them both, then next will be the namespaced state and then etcd and the ephemeral state.

Reply:
alright then I guess I'll go with `infra-provider-ephemeral`.

Reply:
Updated the PR to do that.